### DISCUSSION TITLE
setProp action results in massive type definitions in large projects

### DISCUSSION THREAD

**Discussion by sergioisidoro** [infinitered/ignite#2547] [General]
I've started to hit a typescript limit

```
src/data/stores/RootStore.ts(10,14): error TS7056: The inferred type of this node exceeds the maximum length the compiler will serialize. An explicit type annotation is needed.
```

After a while of debugging, I finally realised that one of the culprits was `setProp` as it was popping out over and over again in the type definitions. Not that it was a recursive issue (it did stop at some point), but having a somewhat deeply nested tree resulted in setProp type annotations to grow VERY rapidly on some cases.

In this case  a single`setProp` in the type definitions has almost 600 lines. 
<img width="723" alt="Screenshot 2023-11-08 at 20 22 49" src="https://github.com/infinitered/ignite/assets/5014629/29ad2865-a009-4a34-9524-f242a0fec541">

So I'm leaving this here as a navigation warning / Improvement idea - use `withSetPropAction` seldomly, or make changes to the type annotations of that action.

**Comment by frankcalise** [comment]
Discussion to continue over at MST repo: https://github.com/mobxjs/mobx-state-tree/discussions/2115
